swift/ql/test/extractor-tests/expressions/methodlookup/PrintAst.expected

```diff
@@ -1,271 +1,706 @@
+#-----| [PatternBindingDecl] var ... = ...
+#-----| getPattern(0) -> [TypedPattern] ... as ...
+
+#-----| [ReturnStmt] return ...
+#-----| getResult() -> [CallExpr] call to UnownedSerialExecutor.init(_:)
+
+#-----| [AccessorDecl] get
+#-----|     InterfaceType = (Bar) -> () -> UnownedSerialExecutor
+#-----| getSelfParam() -> [ParamDecl] self
+#-----| getBody() -> [BraceStmt] { ... }
+
+#-----| [BraceStmt] { ... }
+#-----| getElement(0) -> [ReturnStmt] return ...
+
 #-----| [ModuleDecl] __ObjC
+
 #-----| [ModuleDecl] methodlookup
+
+#-----| [ParamDecl] self
+#-----|     Type = Bar
+
+#-----| [ParamDecl] self
+#-----|     Type = Foo
+
+#-----| [ParamDecl] self
+#-----|     Type = Baz
+
+#-----| [CallExpr] call to UnownedSerialExecutor.init(_:)
+#-----| getFunction() -> [MethodLookupExpr] UnownedSerialExecutor.init(_:)
+#-----| getArgument(0) -> [Argument] : call to ...
+
+#-----| [CallExpr] call to ...
+#-----| getFunction() -> (no string representation)
+#-----| getArgument(0) -> [Argument] : self
+
+#-----| [TypedPattern] ... as ...
+#-----| getSubPattern() -> [NamedPattern] unownedExecutor
+
+#-----| [NamedPattern] unownedExecutor
+
+#-----| [ConcreteVarDecl] unownedExecutor
+#-----|     Type = UnownedSerialExecutor
+#-----| getAccessorDecl(0) -> [AccessorDecl] get
+
+#-----| (no string representation)
+
+#-----| [DeclRefExpr] self
+
+#-----| [DeclRefExpr] self
+
+#-----| [DeclRefExpr] self
+
+#-----| [MethodLookupExpr] UnownedSerialExecutor.init(_:)
+
+#-----| [Argument] : call to ...
+#-----| getExpr() -> [CallExpr] call to ...
+
+#-----| [Argument] : self
+#-----| getExpr() -> [DeclRefExpr] self
+
 methodlookup.swift:
 #    1| [ClassDecl] Foo
-#    2|   getMember(0): [ConstructorDecl] init()
-#    2|       InterfaceType = (Foo.Type) -> () -> Foo
-#    2|     getSelfParam(): [ParamDecl] self
-#    2|         Type = Foo
-#    2|     getBody(): [BraceStmt] { ... }
-#    2|       getElement(0): [ReturnStmt] return
-#    3|   getMember(1): [ConcreteFuncDecl] instanceMethod()
-#    3|       InterfaceType = (Foo) -> () -> ()
-#    3|     getSelfParam(): [ParamDecl] self
-#    3|         Type = Foo
-#    3|     getBody(): [BraceStmt] { ... }
-#    4|   getMember(2): [ConcreteFuncDecl] staticMethod()
-#    4|       InterfaceType = (Foo.Type) -> () -> ()
-#    4|     getSelfParam(): [ParamDecl] self
-#    4|         Type = Foo.Type
-#    4|     getBody(): [BraceStmt] { ... }
-#    5|   getMember(3): [ConcreteFuncDecl] classMethod()
-#    5|       InterfaceType = (Foo.Type) -> () -> ()
-#    5|     getSelfParam(): [ParamDecl] self
-#    5|         Type = Foo.Type
-#    5|     getBody(): [BraceStmt] { ... }
-#    1|   getMember(4): [DestructorDecl] deinit()
-#    1|       InterfaceType = (Foo) -> () -> ()
-#    1|     getSelfParam(): [ParamDecl] self
-#    1|         Type = Foo
-#    1|     getBody(): [BraceStmt] { ... }
```

Comment:
Regarding the flakiness of this test, as you can tell by the diff, after the latest changes (I think it dates back to the schema.py change), this AST output is significantly less hierarchical for some reason (does anyone know why?), and you see random nodes like `DeclRefExpr`, `BraceStmt`, `CallExpr`, `MethodLookupExpr`, `Argument`, and `ReturnStmt` moving around from run to run. I don't think we can stabilize this by excluding a single AST node type from the output.

Reply:
ah, I see. That is most likely a legitimate problem then: with synthesising we have probably removed some parents. This might be a combination of things we remove that we shouldn't, and things we don't remove that we should. As an example, by placing the expression that is the only `SelfApply` argument under `MerhodLookupExpr`, we leave a dangling `Argument` around. We need to remove those by tweaking `ArgumentConstructor.qll`. This explains the dangling `Argument` instances, but I'm not sure about the others.